tests/Functional/ResultCacheTest.php

```diff
@@ -218,11 +221,36 @@ public function testFetchAllSavesCache(): void
             [],
             new QueryCacheProfile(0, 'testcachekey', $layerCache)
         );
-        $result->fetchAllAssociative();
+
+        $fetchAll($result);
 
         self::assertCount(1, $layerCache->fetch('testcachekey'));
     }
 
+    /**
+     * @return iterable<string,list<mixed>>
```

Comment:
Not sure why we didn't use `iterable<string, array{callable}>` here, but this is probably not the place to fix this.

Reply:
I didn't know of this syntax.